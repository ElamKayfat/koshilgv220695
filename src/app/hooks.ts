import { useEffect } from 'react';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { setUserListData } from '../features/UserList/UserListSlice';
import { mockUserList } from './mockData';
import type { RootState, AppDispatch } from './store';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export const useFetchUserListAndSave = () => {
  const dispatch = useDispatch()
  
  useEffect(() => {
    setTimeout(() => {
      dispatch(setUserListData({ userListData: mockUserList }))
    }, 200)
  }, [])
}
