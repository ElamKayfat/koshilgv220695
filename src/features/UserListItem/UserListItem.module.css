.root {
  max-height: 200px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 10px 10px;
}

.root:hover {
  opacity: 0.8;
  cursor: pointer;
}

.wrapper {
  margin-left: 15px;
  display: flex;
  flex-direction: column;
}

.avatar {
  width: 40px;
  height: 40px;
  border-radius: 50%;
}

.button {
  width: 50px;
  height: 25px;
  border: 1px solid lightgray;
  border-radius: 6px;
  color: black;
  background: white;
  cursor: pointer;
}

.button:hover {
  opacity: 0.7;
}

.mainInfoContainer {
  display: flex;
}

.name {
  font-size: 16px;
  font-weight: 500;
}

.nickName {
  font-size: 14px;
  color: darkgray;
}