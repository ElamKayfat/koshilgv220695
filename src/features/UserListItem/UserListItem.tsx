import { UserDataType } from "../UserList/UserListSlice";
import styles from "./UserListItem.module.css";

type UserListItemProps = {
  userData: UserDataType;
  onClick: (userData: UserDataType) => void;
};

function UserListItem(props: UserListItemProps) {
  return (
    <div className={styles.root}>
      <div className={styles.mainInfoContainer}>
        <img
          className={styles.avatar}
          src={props.userData.photo}
          alt={props.userData.name}
        />

        <div className={styles.wrapper}>
          <span className={styles.name}>{props.userData.name}</span>
          <span className={styles.nickName}>{props.userData.nickname}</span>
        </div>
      </div>

      <button
        className={styles.button}
        onClick={() => props.onClick(props.userData)}
        data-testid={`${props.userData.name} view button`}
      >
        View
      </button>
    </div>
  );
}

export default UserListItem;
