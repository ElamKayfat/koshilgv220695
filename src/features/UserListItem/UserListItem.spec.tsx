import { fireEvent, getByTestId, getByText, render } from '@testing-library/react';
import { mockUserList } from '../../app/mockData';
import UserListItem from './UserListItem';

test('Make snapshot. Render with data', () => {
  const { container } = render(<UserListItem userData={mockUserList[0]} onClick={() => undefined}/>);

  expect(container).toMatchSnapshot();
});

test('Make snapshot. Click on view button', () => {
  const mockOnClick = jest.fn()
  const { container } = render(<UserListItem userData={mockUserList[1]} onClick={mockOnClick} />);
  
  fireEvent(
    getByTestId(container, "Brazil Izair view button"),
    new MouseEvent('click', {
      bubbles: true,
      cancelable: true,
    }),
  )
  

  expect(mockOnClick.mock.calls).toHaveLength(1);;
});