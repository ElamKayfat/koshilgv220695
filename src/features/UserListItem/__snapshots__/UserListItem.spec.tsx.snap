// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make snapshot. Render with data 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="mainInfoContainer"
    >
      <img
        alt="Geordan Aaryn"
        class="avatar"
        src="1.jpg"
      />
      <div
        class="wrapper"
      >
        <span
          class="name"
        >
          Geordan Aaryn
        </span>
        <span
          class="nickName"
        >
          @geordanaaryn
        </span>
      </div>
    </div>
    <button
      class="button"
      data-testid="Geordan Aaryn view button"
    >
      View
    </button>
  </div>
</div>
`;
